lib/puttyhosts/puttyhosts_test.go

```diff
@@ -305,3 +309,132 @@ func TestNaivelyValidateHostname(t *testing.T) {
+			desiredOutput: []string{"test"},
+			shouldError:   false,
+		},
+		{
+			input: "*.example.com || test || teleport.test.com",
+			desiredOutput: []string{
+				"*.example.com",
+				"test",
+				"teleport.test.com",
+			},
+			shouldError: false,
+		},
+		{
+			input: "*.example.com || test || teleport.test.com || longstring || *.wow.com",
+			desiredOutput: []string{
+				"*.example.com",
+				"test",
+				"teleport.test.com",
+				"longstring",
+				"*.wow.com",
+			},
+			shouldError: false,
+		},
+		{
+			input:         "*.example.com && port:22",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com && ! *.extrasecure.example.com",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "(*.foo.example.com || *.bar.example.com) && port:0-1023",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "(*.foo.example.com || *.bar.example.com || *.qux.example.com) && port:0-1023",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "((*.foo.example.com || *.bar.example.com || *.qux.example.com) && port:0-1023) || teleport.example.com",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "((*.example.com || lol.example.com && port:22) && port:1024) || test.teleport.com || teleport.test.com",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com || lol.example.com | test.teleport.com",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com && lol.example.com || test.teleport.com",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com || lol.example.com || | test.teleport.com",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com || lol.example.com || | test.teleport.com || ",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com || lol.example.com || test.teleport.com ",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         " *.example.com || lol.example.com",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com || lol.example.com || test.teleport.com || \"\"",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+		{
+			input:         "*.example.com || lol.example.com || .",
+			desiredOutput: []string(nil),
+			shouldError:   true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.input, func(t *testing.T) {
+			testResult, err := CheckAndSplitValidityKey(tt.input, "TeleportHostCA-testcluster.example.com")
+			if err != nil {
+				t.Log(err)
+			}
+			require.Equal(t, tt.desiredOutput, testResult)
+			if tt.shouldError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
```

Comment:
nit: I would normally write it this way, I don't think there's a need to log expected errors or to check the result if there was an error
```suggestion
			if tt.shouldError {
				require.Error(t, err)
				return
			}
			require.NoError(t, err)
			require.Equal(t, tt.desiredOutput, testResult)
```